rqpy/process/_process_iv_didv.py

```diff
@@ -79,146 +77,215 @@ def _process_ivfile(filepath, chans, detectorid, rfb, loopgain, binstovolts,
-        print(f'------------------\n Processing dumps in file: {filepath} \n------------------')
-    
+        print(f'\n\n============================')
+        print(f'Processing dumps in file: {filepath}')
+        print(f'============================\n')
     
     if isinstance(chans, str):
         chans = [chans]
     
-    detnum = int(detectorid[-1])
-    
+    detnum = int(detectorid[1:])
     nchan = len(chans)
     
+
+    # Initialize data list
     data_list = []
+
+    # serie number (a bit sketchy...)
     if filepath[-1] == '/':
         seriesnum = filepath[:-1].split('/')[-1]
     else:
         seriesnum = filepath.split('/')[-1]
-    reader = rawdata.DataReader()
-    settings_path = glob(f"{filepath}*")[0]
-    reader.set_filename(settings_path)
-
-    if not lgcHV:
-        odb_list = [f"/Detectors/Det0{detnum}/Readback/TestSignal/Amplitude (mV)",
-                    f"/Detectors/Det0{detnum}/Readback/TestSignal/Frequency (Hz)",
-                    f"/Detectors/Det0{detnum}/Readback/TestSignal/GeneratorEnable",
-                    f"/Detectors/Det0{detnum}/Readback/TestSignal/QETTestEnable",
-                    f"/Detectors/Det0{detnum}/Readback/TestSignal/QETTestSelect"]
-    else:
-        odb_list = [f"/Detectors/Det0{detnum}/Readback/TestSignal/Amplitude (mV)",
-                    f"/Detectors/Det0{detnum}/Readback/TestSignal/Frequency (Hz)",
-                    f"/Detectors/Det0{detnum}/Readback/TestSignal/GeneratorEnable",
-                    f"/Detectors/Det0{detnum}/Readback/TestSignal/QETTestEnable",
-                    f"/Detectors/Det0{detnum}/Readback/TestSignal/QETTestSelect[0]",
-                    f"/Detectors/Det0{detnum}/Readback/TestSignal/QETTestSelect[1]"]
-    reader.set_odb_list(odb_list)
-    odb_dict = reader.get_odb_dict()
-    
-    lgcGenEnable = odb_dict[odb_list[2]] #True/False
-    lgcQETEnable = odb_dict[odb_list[3]] #True/False
-    qetChanSelect0 =  odb_dict[odb_list[4]] #channel number
-    if lgcHV:
-        qetChanSelect1 =  odb_dict[odb_list[5]] #channel number
-    else:
-        qetChanSelect1 =  odb_dict[odb_list[4]] #channel number
-        
-    ### Load traces and channel names in order of loaded traces
-    events = getRawEvents(filepath, "", channelList=chans, detectorList=[detnum], outputFormat=3)
-    channels = events[detectorid]["pChan"] # we use the returned channels rather than the user 
-                                           # provided channel list because the order returned 
-                                           # from getRawEvents() is not nessesarily in the 
-                                           # expected order
+
+
+    # Load traces for user provided detector and channels
+    events = []
+    try:
+        events = midasio.getRawEvents(
+            filepath, "", channelList=chans, detectorList=[detnum], outputFormat=3,
+        )
+    except:
+        print('ERROR in process_iv_didv: Unable to get traces!')
+        return
```

Comment:
Maybe rather than printing and returning, this should raise an error. That way the user can catch this and find the bad file. 

Also, does getRawEvents() raise any errors? If not, this try statement will never fail anyway.

Reply:
Yes, rawio functions will fail if problems with raw data.  I agree it is probably better to let it  fails by just removing try/except. 
I had the vague idea to let it continue processing so that not everything is wasted when multiple series are  being processed  together, however  some codes would  need to be added for bookkeeping purpose.

Reply:
I don't have any issue with the try/except. I was just saying that it could be more helpful for the case that it fails, have it raise an error in the except rather than a print statement. But either way, I do agree that it does make sense to continue processing even for bad data. However, in its current form I don't think it will work, or if it does, its behavior is unclear. This function returns a nested list `data_list` where each entry is a list: `data = [chan, seriesnum, fs, qetbias, sgamp, sgfreq, offset, offset_err, 
                    f, psd, avgtrace, didvmean, didvstd, datatype, cut_eff, cut, cut_pass]`. But when it fails it just returns. So I'm unsure how the rest of the processing will handle that if it just gets an empty return. 

I think if we want the processing to continue, in the case that it fails, it should return the same `data_list` but with everything set to None, or 'fail' or something?

Reply:
> Yes, rawio functions will fail if problems with raw data. I agree it is probably better to let it fails by just removing try/except.

Looking through the `rawio` functions, I don't think this is necessarily the case. That is, the `rawio` functions use try/except statements like so (e.g. in [getDetectorSettings](http://titus.stanford.edu:8080/git/blob/?f=rawio/IO.py&r=DataHandling/pyRawIO.git&h=master#L204)):

```python
try:
    # try something
except:
    print("Error!")
    return
```

Thus, an error is never raised, as opposed to if it was written as:

```python
try:
    # try something
except:
    raise ValueError("Invalid input!")
```

So, in `rqpy`, when we call a function that has an error in one of these try/except statements, then we won't actually trigger the exception. So, it's not entirely clear to me when these exceptions will be raised due to the use of print statements rather than raising errors.

Also, I think that if we do want to write the code so that it continues in the case of an error, we should use warnings rather than print statements. It still prints out the warning message, but will also tell the user what line of code the warning occurs in, as well as has a nice light red background!

Reply:
Yes I agree, a warning would be a better way to handle that case

Reply:
Hi Sam, getRawEvents will definitely fail if there is a problem with raw data. There is no try/except when getting the raw traces.  For the detector settings, there is also no try/except in the python code but indeed it may not fail if the ODB path doesn't exist. In general, I did not want failure if ODB path doesn't exist, only if there is an actual problem with raw data. 

In any cases, I would suggest we add raise ValueError in the RQpy code. Implementing more code to properly continue processing is I think beyond the work on this branch.

Reply:
> In any cases, I would suggest we add raise ValueError in the RQpy code. Implementing more code to properly continue processing is I think beyond the work on this branch.

Agreed!

Reply:
I've made the changes (plus some white space fixes to make things a bit more PEP8 compliant), I'll commit them shortly.